Analysis/Tasks/PWGCF/o2femtodream/include/o2femtodream/FemtoDreamTrackSelection.h

```diff
@@ -0,0 +1,480 @@
+    }
+    if (nEtaMaxSel > 0) {
+      mHistogramRegistry->add("TrackCutsQA/etaMax", "; Cut; #eta", kTH2F, {{nEtaMaxSel, 0, static_cast<double>(nEtaMaxSel)}, {1000, -1, 1}});
+    }
+    if (nTPCnMinSel > 0) {
+      mHistogramRegistry->add("TrackCutsQA/tpcnClsMin", "; Cut; TPC Cluster", kTH2F, {{nTPCnMinSel, 0, static_cast<double>(nTPCnMinSel)}, {163, 0, 163}});
+    }
+    if (nTPCfMinSel > 0) {
+      mHistogramRegistry->add("TrackCutsQA/tpcfClsMin", "; Cut; TPC ratio findable", kTH2F, {{nTPCfMinSel, 0, static_cast<double>(nTPCfMinSel)}, {100, 0.5, 1.5}});
+    }
+    if (nTPCcMinSel > 0) {
+      mHistogramRegistry->add("TrackCutsQA/tpcnRowsMin", "; Cut; TPC crossed rows", kTH2F, {{nTPCcMinSel, 0, static_cast<double>(nTPCcMinSel)}, {163, 0, 163}});
+    }
+    if (nTPCsMaxSel > 0) {
+      mHistogramRegistry->add("TrackCutsQA/tpcnSharedMax", "; Cut; TPC shared clusters", kTH2F, {{nTPCsMaxSel, 0, static_cast<double>(nTPCsMaxSel)}, {163, 0, 163}});
+    }
+    if (nDCAxyMaxSel > 0) {
+      mHistogramRegistry->add("TrackCutsQA/dcaXYMax", "; Cut; DCA_{xy} (cm)", kTH2F, {{nDCAxyMaxSel, 0, static_cast<double>(nDCAxyMaxSel)}, {51, -3, 3}});
+    }
+    if (nDCAzMaxSel > 0) {
+      mHistogramRegistry->add("TrackCutsQA/dcaZMax", "; Cut; DCA_{z} (cm)", kTH2F, {{nDCAzMaxSel, 0, static_cast<double>(nDCAzMaxSel)}, {51, -1.5, 1.5}});
+    }
+    if (nPIDnSigmaSel > 0) {
+      int nSpecies = mPIDspecies.size();
+      mHistogramRegistry->add("TrackCutsQA/pidCombnsigmaMax", "; #it{n}_{#sigma, comb.}; Cut", kTH2F, {{nPIDnSigmaSel * nSpecies, 0, static_cast<double>(nSpecies * nPIDnSigmaSel)}, {60, 0, 6}});
+      mHistogramRegistry->add("TrackCutsQA/pidTPCnsigmaMax", "; #it{n}_{#sigma, TPC}; Cut", kTH2F, {{nPIDnSigmaSel * nSpecies, 0, static_cast<double>(nSpecies * nPIDnSigmaSel)}, {121, -6, 6}});
+    }
+  }
+}
+
+template <typename T>
+inline auto FemtoDreamTrackSelection::getNsigmaTPC(T const& track, o2::track::PID pid)
+{
+  switch (pid) {
+    case o2::track::PID::Electron:
+      return track.tpcNSigmaEl();
+      break;
+    case o2::track::PID::Muon:
+      return track.tpcNSigmaMu();
+      break;
+    case o2::track::PID::Pion:
+      return track.tpcNSigmaPi();
+      break;
+    case o2::track::PID::Kaon:
+      return track.tpcNSigmaKa();
+      break;
+    case o2::track::PID::Proton:
+      return track.tpcNSigmaPr();
+      break;
+    case o2::track::PID::Deuteron:
+      return track.tpcNSigmaDe();
+      break;
+    default:
+      return 999.f;
+      break;
+  }
+}
+
+template <typename T>
+inline auto FemtoDreamTrackSelection::getNsigmaTOF(T const& track, o2::track::PID pid)
+{
+  /// skip tracks without TOF signal
+  /// \todo not sure what the error flags mean...
+  if (track.tofSignal() <= 0.f || std::abs(track.tofSignal() - 99998) < 0.01 || std::abs(track.tofSignal() - 99999) < 0.01) {
```

Comment:
why these comparisons with subtraction of 99998 and 99999?

Reply:
these appear to be some error flags from TOF PID. Maybe an isValid() function would be helpful here?

Reply:
Isn't the hasTOF() enough?
https://github.com/AliceO2Group/AliceO2/blob/dev/Framework/Core/include/Framework/AnalysisDataModel.h#L213

Reply:
in the case of 99998 and 99999 the TOF signal is clearly larger than 0, so I don't think this will be enough.